### PR TITLE
Add support for PSR clock

### PR DESCRIPTION
**Pull request by X-Coder264**
## Subject

I am targeting this branch, because it adds a new feature without a BC-break.

## Changelog

```markdown
### Added
- Support for PSR ClockInterface
```

### CODE REVIEW

File: src/EventListener/LogRevisionsListener.php [hunk 1]
```diff
@@ -323,15 +327,17 @@ private function getManyToManyRelations(EntityManagerInterface $em, object $enti
...
             $conn->insert(
                 $this->config->getRevisionTableName(),
                 [
-                    'timestamp' => date_create('now'),
+                    'timestamp' => $now,
                     'username' => $this->config->getCurrentUsername(),
                 ],
                 [
-                    Types::DATETIME_MUTABLE,
+                    Types::DATETIME_IMMUTABLE,
```
Comment (VincentLanglet) [inline src/EventListener/LogRevisionsListener.php:340]: Isn't this change a BC-break ? I dunno what it imply.

File: tests/BaseTest.php [hunk 2]
```diff
@@ -171,12 +172,22 @@ protected function getAuditManager(): AuditManager
...
         $auditConfig->setUsernameCallable(static fn (): string => 'beberlei');
 
-        $auditManager = new AuditManager($auditConfig);
+        $auditManager = new AuditManager($auditConfig, $this->getClock());
         $auditManager->registerEvents($this->_getConnection()->getEventManager());
 
         return $this->auditManager = $auditManager;
     }
 
+    protected function getClock(): ClockInterface
```
Comment (VincentLanglet) [inline tests/BaseTest.php]: This should return null by default instead to test that we correctly work without clock.
```suggestion
    protected function getClock(): ?ClockInterface
    {
         return null;
    }
```

**Review by VincentLanglet — CHANGES_REQUESTED** [review]

File: tests/BaseTest.php (hunk 2 above)
Comment (X-Coder264) [reply tests/BaseTest.php]: Done

File: src/EventListener/LogRevisionsListener.php (hunk 1 above)
Comment (X-Coder264) [reply src/EventListener/LogRevisionsListener.php:340]: It's not a BC break. The query either executes or it doesn't (it obviously executes fine as all the tests are passing).

All the types do is tell DBAL which `\Doctrine\DBAL\Types\Type` child class to use to get the value to bind to the query, more specifically in this case whether to call `\Doctrine\DBAL\Types\DateTimeType::convertToDatabaseValue` or `\Doctrine\DBAL\Types\DateImmutableType::convertToDatabaseValue`.

**Review by VincentLanglet — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Thanks

**Pull request merged**